services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchFailuresSpec.groovy

```diff
@@ -63,8 +61,8 @@ class SwitchFailuresSpec extends BaseSpecification {
         when: "Two neighbouring switches of the flow go down simultaneously"
         lockKeeperService.knockoutSwitch(isl.srcSwitch.dpId.toString())
         lockKeeperService.knockoutSwitch(isl.dstSwitch.dpId.toString())
-        def timeIslBroke = System.currentTimeMillis()
-        def untilIslShouldFail = { timeIslBroke + discoveryTimeout * 1000 - System.currentTimeMillis() }
+        def timeSwitchesBroke = System.currentTimeMillis()
```

Comment:
It is not true. It is used every time you call untilIslShouldFail() closure. If I put it inside the closure it will take current time every time (instead of using exact time when switches were broken)